### PR TITLE
sys: util: Add IF_ENABLED() macro

### PR DESCRIPTION
**Pull request by nordic-krch**
Added macro for code inserting based on configuration flag. This macro is wrapper around `COND_CODE_1()`. Macro can be seen as an alternative to ifdef

Example:
```
#if defined(CONFIG_FOO) && (CONFIG_FOO == 1)
u32_t foo;
#endif
```
can be replaced with:
```
IF_ENABLED(CONFIG_FOO, (u32_t foo;))
```

Compared to standard ifdef approach, `IF_ENABLED()` can also be used inside other macros, like:
```
#define FOO_DEFINE(_name) \
   struct foo { \
       IF_ENABLED(CONFIG_FOO_FEATURE, (.value = CONFIG_FOO_VALUE,) \
   }
```

PR contains:
- macro implementation
- test for new macro
- replacing all applicable `COND_CODE_1` usages with `IF_ENABLED`, cleanup around `COND_CODE_1` usage.

### CODE REVIEW

File: drivers/serial/uart_esp32.c [hunk 1]
```diff
@@ -474,68 +474,69 @@ static const struct uart_driver_api uart_esp32_api = {
...
-DEVICE_AND_API_INIT(uart_esp32_##idx,							      \
-		    DT_INST_##idx##_ESPRESSIF_ESP32_UART_LABEL,				      \
-		    uart_esp32_init,							      \
-		    &uart_esp32_data_##idx,						      \
-		    &uart_esp32_cfg_port_##idx,						      \
-		    PRE_KERNEL_1,							      \
-		    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,					      \
-		    &uart_esp32_api);							      \
-											      \
+#endi
```
Comment (tbursztyka) [inline drivers/serial/uart_esp32.c]: typo

File: drivers/serial/uart_esp32.c [hunk 2]
```diff
@@ -474,68 +474,69 @@ static const struct uart_driver_api uart_esp32_api = {
...
+		.tx_out = U##idx##TXD_OUT_IDX,				       \
+		.rx_in = U##idx##RXD_IN_IDX,				       \
+		.rts_out = U##idx##RTS_OUT_IDX,				       \
+		.cts_in = U##idx##CTS_IN_IDX,				       \
+	},								       \
+									       \
+	.pins = {							       \
+		.tx = DT_INST_##idx##_ESPRESSIF_ESP32_UART_TX_PIN,	       \
+		.rx = DT_INST_##idx##_ESPRESSIF_ESP32_UART_RX_PIN,	       \
+		COND_CODE_1(DT_INST_##idx##_ESPRESSIF_ESP32_UART_HW_FLOW_CONTROL,\
```
Comment (tbursztyka) [inline drivers/serial/uart_esp32.c]: so you do not replace this by IF_ENABLED?

Comment (nordic-krch) [reply drivers/serial/uart_esp32.c]: right, this could be replaced as well.

**Review by masz-nordic — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by anangl** [comment]
There are two more occurrences of `COND_CODE_1` that you could replace with `IF_ENABLED` after rebasing this PR on current master. See `nrf_power_clock.c`, [line 354](https://github.com/zephyrproject-rtos/zephyr/blob/e9cf3e9963aac543717ec7c8fd5c23a22f743f87/drivers/clock_control/nrf_power_clock.c#L354) and [line 360](https://github.com/zephyrproject-rtos/zephyr/blob/e9cf3e9963aac543717ec7c8fd5c23a22f743f87/drivers/clock_control/nrf_power_clock.c#L360).

**Comment by nordic-krch** [comment]
@anangl thanks, applied. I've also used `IF_ENABLED` for setting start/stop handlers.

**Review by anangl — APPROVED** [review]

**Pull request merged**